index.js

```diff
@@ -337,7 +337,7 @@
                 /* Optimal order of selectors for checking. */
                 var selector_elements = ['year', 'holiday', 'month', 'monthday', 'week', 'weekday'];
                 for (var selector_ind in selector_elements) {
-                    if (selectors[selector_elements[selector_ind]].length > 0) {
+                    if (selector_elements.hasOwnProperty(selector_ind) && selectors[selector_elements[selector_ind]].length > 0) {
```

Comment:
There is also [xo](https://github.com/sindresorhus/xo) which I wanted to take a closer look at. I guess either one or even multiple combined should do the job just fine. Still better than none of the above and only uglifyjs with its warnings.

> So... in this case I suggest to make a PR first to put in place ESLint & fix existing code formatting, then rebase & push again on this PR, WDYT?

SGTM :wink: